internal/pkg/poster/poster.go

```diff
@@ -70,148 +81,146 @@ func (p *Poster) StopPoSting() {
 }
 
 func (p *Poster) startPoStIfNeeded(ctx context.Context, newHead block.TipSet) error {
-	// TODO: Window PoSt (#4010)
-	//p.postMutex.Lock()
-	//defer p.postMutex.Unlock()
-	//
-	//if p.postCancel != nil {
-	//	// already posting
-	//	return nil
-	//}
-	//
-	//tipsetHeight, err := newHead.Height()
-	//if err != nil {
-	//	return err
-	//}
-	//
-	//root, err := p.chain.GetTipSetStateRoot(ctx, newHead.Key())
-	//if err != nil {
-	//	return err
-	//}
-	//
-	//stateView := p.stateViewer.StateView(root)
-	//provingPeriodStart, _, _, err := stateView.MinerProvingPeriod(ctx, p.minerAddr)
-	//if err != nil {
-	//	return nil
-	//}
-	//
-	//if provingPeriodStart > tipsetHeight {
-	//	// it's not time to PoSt
-	//	return nil
-	//}
-	//
-	//ctx, p.postCancel = context.WithCancel(ctx)
-	//go p.doPoSt(ctx, stateView, provingPeriodStart, newHead.Key())
-	//
+	p.postMutex.Lock()
+	defer p.postMutex.Unlock()
+
+	if p.postCancel != nil {
+		// already posting
+		return nil
+	}
+
+	tipsetHeight, err := newHead.Height()
+	if err != nil {
+		return err
+	}
+
+	root, err := p.chain.GetTipSetStateRoot(ctx, newHead.Key())
+	if err != nil {
+		return err
+	}
+
+	stateView := p.stateViewer.StateView(root)
+	index, _, _, challenge, err := stateView.MinerDeadlineInfo(ctx, p.minerAddr, tipsetHeight)
```

Comment:
This will need to handle the incomplete proving period right after genesis, when no proofs are due. The actors code now specifies a `ProvingPeriodStart` which, if it's in the future, means nothing to do.

Reply:
I'm unclear what this means. Currently the miner starts with no sectors in its deadlines. Those don't get added until the cron job sets up the next proving period. With no sectors due, the miner doesn't PoSt in that incomplete proving period. Is this not the correct behavior? Does the genesis block start with miners that have sectors due that are supposed to be ignored?

Reply:
You're right, the staging of genesis sectors in NewSectors handles this for us.